### DISCUSSION TITLE
Stress sensor not working in applet

### DISCUSSION THREAD

**Discussion by zdunex25** [zepp-health/zeppos-forum#253] [Developer Feedback]
>import { Stress } from '@zos/sensor'
>
>const stress = new Stress()
>const { value, time } = stress.getCurrent();
>
>createWidget(widget.TEXT, {
>            x: 0,
>            y: 45,
>            w: 466,
>            h: 60,
>            color: 0xf07333,
>            text_size: 36,
>            align_h: align.CENTER_H,
>            align_v: align.TOP,
>            text_style: text_style.NONE,
>            text: \``${value}`\`
>        });`
>
>createWidget(widget.TEXT, {
>            x: 0,
>            y: 95,
>            w: 466,
>            h: 120,
>            color: 0xffffff,
>            char_space: -1,
>            text_size: 90,
>            align_h: align.CENTER_H,
>            align_v: align.TOP,
>            text_style: text_style.NONE,
>            text: stress.getCurrent()
>        });

It returns either:
**[object object]** when I use _stress.getCurrent()_
or
**0** when I use _\`${value}\`_

Same behavior in Showcase sample app, no matter if you try on simulator or a real device.
"data:user.hd.stress" has been added.

@MyoungXUE @silver-zepp

**Comment by silver-zepp** [comment]
I think it's a known issue, at least we had this information available in the [1.0 docs](https://docs.zepp.com/docs/1.0/reference/device-app-api/hmSensor/sensorId/STRESS) but the caution block seems to be missing in the v2+ docs @MyoungXUE.
Honestly, I wouldn't rely much on this _sensor_, as its algorithm is flawed @zdunex25

**Reply by zdunex25** [reply]
I see the note now, but it's been a while I guess, and it seems like **temporary** became **permanent** :/
Is it fixable or even going to be fixed (I mean gtr4 if it matters) @MyoungXUE

**Comment by MyoungXUE** [comment]
@zdunex25 

Hi, the `stess.getCurrent()` method returns an object, and displaying the object directly as a text gives you `[object Object]`

You get `0` because the stress sensor of the watch hasn't calculated the stress value yet, you need to turn on the stress monitoring in the health monitoring of the watch and keep wearing it for a while before you can get the stress measurement value.

**Reply by zdunex25** [reply]
![IMG_20231007_053315.jpg](https://github.com/zepp-health/zeppos-forum/assets/1674725/f1ce5539-0a82-486a-874e-199209e753f1)

The problem is that I do have stress measurement enabled on my watch, yet it still shows 0.

If it really requires a measure wouldn't it be better to make it work like spo2 or hr sensor, where you can call start/stop and get retCode (status that tells what is happening)?
@MyoungXUE

**Reply by MyoungXUE** [reply]
@zdunex25 
Can you provide me with the device model and firmware version and I'll come and try to reproduce the problem?

**Reply by zdunex25** [reply]
@MyoungXUE ok, so device is gtr4 and fw is 3.23.3.1, the other thing i tried was to replace **stress.getCurrent()** with **stress.getCurrent().value**

**Reply by zdunex25** [reply]
@MyoungXUE have you had a chance to verify?

**Reply by MyoungXUE** [reply]
@zdunex25  Hello, I have confirmed this issue. Sorry for not replying to you. We will fix this issue in the next OTA, but it may take some time. However, I believe there will be a big surprise after waiting.

**Reply by khayargoli** [reply]
i get the same issue in gtr4, i always get the getCurrent() value as 255, why?

also the onChange method never triggers
